### PR TITLE
Usersetingtest

### PR DESCRIPTION
**Pull request by songkaiape**
1. update  debug proxy to support user setting
2. set default loglevel to fine
3. fix a argstest bug, utf-16  make the case fail but if still pass so I add assert count to make sure all assert have been hit.

### CODE REVIEW

File: src/debug/args-test.js [hunk 1]
```diff
@@ -86,6 +87,7 @@ class ArgsTest {
         const breakpointFile = path.join(engine.cwd, this.sourcePath, 'ArgsTest.java');
         const expectedLine = 20;
         const outputList = [];
+        var assertCount=0;
```
Comment (andxu) [inline src/debug/args-test.js]: var => let

let assertCount = 0; // be careful about the code style.

File: src/debug/args-test.js [hunk 2]
```diff
@@ -100,20 +102,26 @@ class ArgsTest {
                         console.log('----->', await engine.variables(variable.variablesReference));
                     }
                     if (variable.name === 'args') {
-                        variable.type.should.equal('java.lang.String[]');
-                        utils.shouldMatch(variable.value, /^java.lang.String\[3]\s+\(id=\d+\)$/g);
+                        variable.type.should.equal('String[]');
+                        utils.shouldMatch(variable.value, /^String\[3]\s+\(id=\d+\)$/g);
+                        assertCount+=1;
```
Comment (andxu) [inline src/debug/args-test.js]: assertCount++;

File: src/debug/args-test.js (hunk 1 above)
Comment (songkaiape) [reply src/debug/args-test.js]: Got it

File: src/debug/test-utils.js [hunk 3]
```diff
@@ -76,8 +76,14 @@ export async function createDebugEngine(DATA_ROOT, LANGUAGE_SERVER_ROOT, LANGUAG
     if (!fs.isDirectorySync(LANGUAGE_SERVER_ROOT)) {
         throw new Error(`${LANGUAGE_SERVER_ROOT} doesn't exist.`);
     }
-
-    const promise1 = startDebugServer(DATA_ROOT, config.logLevel || 'FINE');
+    var defaultSettings={
```
Comment (andxu) [inline src/debug/test-utils.js]: Turn off the logLevel from FINE to INFO for some tests which has very large vscode sesssions. See my previous code https://github.com/vscjavaci/java-debug-ci/blob/master/src/debug/stacktrace-test.js


  get logLevel() {
        return "INFO";
    }

File: src/debug/args-test.js [hunk 4]
```diff
@@ -87,7 +87,7 @@ class ArgsTest {
         const breakpointFile = path.join(engine.cwd, this.sourcePath, 'ArgsTest.java');
         const expectedLine = 20;
         const outputList = [];
-        var assertCount=0;
+        let assertCount=0;
```
Comment (andxu) [inline src/debug/args-test.js]: add space to the left and right of = operator.

Comment (andxu) [reply src/debug/args-test.js]: https://github.com/elierotenberg/coding-styles/blob/master/es6.md 
<pre>
// bad
const a = b+c;
const a= b + c;

// good
const a = b + c;
</pre>

Comment (songkaiape) [reply src/debug/args-test.js]: OK, I will be care full with this and update the code

**Review by andxu — APPROVED** [review]

**Pull request merged**